src/Generator/Generators/CSharp/CSharpSources.cs

```diff
@@ -275,7 +275,8 @@ private void GenerateClassTemplateSpecializationInternal(Class classTemplate)
             if (generateNested)
             {
                 foreach (var nestedClass in classTemplate.Classes.Where(c => !c.IsDependent).Union(
-                    generated.First().Classes.Where(c => !c.IsDependent)))
+                    (generated.FirstOrDefault(s => !s.Ignore) ?? generated.First()).Classes.Where(
```

Comment:
This nested usage of LINQ is hard to read. Can you maybe split up these nested LINQ into locals with a comment explaining what we're doing here?